training/dataset.py

```diff
@@ -377,3 +398,15 @@ def __iter__(self):
                 yield self.buckets[(f, h, w)]
                 del self.buckets[(f, h, w)]
                 self.buckets[(f, h, w)] = []
+
+        if self.drop_last:
+            return
+
+        for fhw, bucket in list(self.buckets.items()):
+            if len(bucket) == 0:
+                continue
+            if self.shuffle:
+                random.shuffle(bucket)
```

Comment:
Could also fix the seed here?

Reply:
set_seed already does this in the training and preparation scripts